src/tab.js

```diff
@@ -46,33 +46,34 @@ function tab({shift = false, focusTrap} = {}) {
     })
     .map(({el}) => el)
 
-  if (shift) orderedElements.reverse()
-
   // keep only the checked or first element in each radio group
```

Comment:
This premise might be flawed.

```html
<button>Foo</button>
<label><input type='radio' name='myRadio'/>Radio1</label>
<label><input type='radio' name='myRadio'/>Radio2</label>
<button>Bar</button>
<label><input type='radio' name='myRadio'/>Radio3</label>
<button>Baz</button>
```

(Tested in Chrome:)
Until one of the radio buttons is checked, the focus jumps `Foo -> Radio1 -> Bar -> Radio3 -> Baz` and with shift `Baz -> Radio3 ->Bar -> Radio2 -> Foo`

While you cannot focus an unchecked radio button with keyboard/mouse, when another radio of that group is checked, focus can still be set by a script.

```js
document.querySelector('button').addEventListener('click', () => document.querySelector('input').focus())
```

If you check `Radio3`, then click `Foo`, and then press `tab` the focus will be on `Bar`. Focused element should never be pruned, even if another radio button is checked.

Reply:
I got you point. Maybe we should make more check on this. For example if the current element in the loop is the one with the check and the previous has the focus we should not delete that element.
We need to consider also a case where there are more radios with different names

Reply:
hi @ph-fritsche I have reimplemented that part. There are some checks to make this works. Now the implementation is little tricky, maybe it can be made in a simpler way. Could you please take a look ? Thank you

Reply:
see https://github.com/marcosvega91/user-event/pull/1

Reply:
much clearer. Do you want to push your changes here? For me you can eventually close this PR and open a new one